src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -77,37 +76,15 @@ public struct MetricPoint
 
         public double DoubleValue { get; internal set; }
 
-        internal MetricPointStatus MetricPointStatus { get; private set; }
-
-        public long[] GetBucketCounts()
-        {
-            if (this.aggType == AggregationType.Histogram)
-            {
-                return this.histogramMeasurements.AggregatedBucketCounts;
-            }
-            else
-            {
-                throw new NotSupportedException($"{nameof(this.GetBucketCounts)} is not supported for this metric type.");
-            }
-        }
+        public readonly HistogramMeasurements HistogramMeasurements { get; }
```

Comment:
Just typing my understanding:

MetricPoint.GetHistogramSum, GetHistogramCount is used to retrieve sum/count of Histogram. (always expected for a Histogram)

For Buckets (optional), it'll be retrieved from the HistogramBuckets.

Reply:
I think that might change after the struct layout stuff I'm looking at. I think leave it like this for now and then we can revisit?

Reply:
yes exactly.